# Spotify

Nada Trabelsi  
2024-03-08

## Loading the Dataset

In [1]:
import pandas as pd
spotify = pd.read_csv('https://bcdanl.github.io/data/spotify_all.csv')
spotify.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 198005 entries, 0 to 198004
Data columns (total 7 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   pid            198005 non-null  int64 
 1   playlist_name  198005 non-null  object
 2   pos            198005 non-null  int64 
 3   artist_name    198005 non-null  object
 4   track_name     198005 non-null  object
 5   duration_ms    198005 non-null  int64 
 6   album_name     198005 non-null  object
dtypes: int64(3), object(4)
memory usage: 10.6+ MB

## Variable Description

- pid: playlist ID; unique ID for playlist
- playlist_name: a name of playlist
- pos: a position of the track within a playlist (starting from 0)
- artist_name: name of the track’s primary artist
- track_name: name of the track
- duration_ms: duration of the track in milliseconds
- album_name: name of the track’s album

## Filtering

I will filter for one of my favorite artists: Shakira.

In [2]:
shakira = spotify[ spotify["artist_name"] == "Shakira"]
shakira

174 rows × 7 columns

## Indexing and Sorting

In [3]:
shakira= (
          shakira
          .set_index("pid")
          .sort_index(ascending=True)
)
shakira.sort_values(by = ["playlist_name", "pos"],
                    ascending = True)

174 rows × 6 columns

## Counting

In [4]:
shakira.nunique()

playlist_name    116
pos               86
artist_name        1
track_name        35
duration_ms       36
album_name        12
dtype: int64

There are 116 unique playlists that have atleast 1 Shakira song in it.

In [5]:
shakira['playlist_name'].value_counts()

playlist_name
💃🏽              9
musica          7
spanish jams    6
GB              6
Party           4
               ..
Taylor swift    1
Gabriel         1
weekend         1
misc.           1
everything      1
Name: count, Length: 116, dtype: int64

The playlist with the most Shakira songs is “💃” with 9 songs included.

In [6]:
shakira["track_name"].value_counts()

track_name
Hips Don't Lie                                                                   53
Chantaje                                                                         18
La Tortura                                                                       13
Waka Waka (This Time for Africa) (The Official 2010 FIFA World Cup (TM) Song)    12
Try Everything - From "Zootropolis"                                               9
Can't Remember to Forget You                                                      9
Whenever, Wherever                                                                8
Me Enamoré                                                                        7
Rabiosa                                                                           5
Perro Fiel                                                                        4
Loca                                                                              4
Addicted to You                                                  

The most popular song (added to playlist the most times) is “Hips Don’t
Lie”.